File: lib/src/com/zwb/commonlibs/http/HttpHelper.java
package com.zwb.commonlibs.http;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

import com.zwb.commonlibs.utils.LogUtils;

import java.io.ByteArrayOutputStream;
import java.io.IOException;
import java.io.InputStream;
import java.io.UnsupportedEncodingException;
import java.net.HttpURLConnection;
import java.net.URL;
import java.net.URLEncoder;
import java.util.HashMap;
import java.util.Map;

/**
 * Created by zwb on 2015/4/11.
 * 基于URLConnection的Http实现
 */
public class HttpHelper {
    private String encoding = "utf-8";

    private static String inputStream2String(InputStream is, String encoding) throws IOException {
        ByteArrayOutputStream baos = new ByteArrayOutputStream();
        int i;
        while ((i = is.read()) != -1) {
            baos.write(i);
        }
        return baos.toString(encoding);
    }

    public static boolean getNetworkState(Context context) {
        ConnectivityManager manager = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo networkInfo = manager.getActiveNetworkInfo();
        return networkInfo != null && networkInfo.getState() == NetworkInfo.State.CONNECTED;
    }

    private static String buildUrl(String url, HttpMap map) {
        return map != null ? url + "?" + map : url;
    }

    public void setEncoding(String encoding) {
        this.encoding = encoding;
    }

    public String getGetResult(String url) throws IOException {
        String string = inputStream2String(getGetInputStream(url), encoding);
        LogUtils.i(this, string);
        return string;
    }

    public String getGetResult(String url, HttpMap map) throws IOException {
        return getGetResult(buildUrl(url, map));
    }

    public InputStream getGetInputStream(String url) throws IOException {
        HttpURLConnection urlConnection = (HttpURLConnection) new URL(url).openConnection();
        setupURLConnection(urlConnection);
        urlConnection.connect();
        return urlConnection.getInputStream();
    }

    public String getPostResult(String url, HttpMap data) throws IOException {
        String string = inputStream2String(getPostInputStream(url, data), encoding);
        LogUtils.i(this, string);
        return string;
    }

    public InputStream getPostInputStream(String url, HttpMap data) throws IOException {
        HttpURLConnection urlConnection = (HttpURLConnection) new URL(url).openConnection();
        setupURLConnection(urlConnection);
        if (data != null) {
            urlConnection.setDoOutput(true);
            urlConnection.setDoInput(true);
            urlConnection.getOutputStream().write(data.toString().getBytes());
            urlConnection.getOutputStream().flush();
            urlConnection.getOutputStream().close();
        }
//        LogUtils.i(this, "responseCode=" + urlConnection.getResponseCode());
        return urlConnection.getInputStream();
    }

    protected void setupURLConnection(HttpURLConnection urlConnection) {
        urlConnection.setRequestProperty("content-type", "application/x-www-form-urlencoded");
        urlConnection.setConnectTimeout(10000);
        urlConnection.setReadTimeout(10000);
    }

    public static class HttpMap {
        private Map<String, String> map = new HashMap<>();

        public HttpMap addParam(String name, Object value) {
            map.put(name, value == null ? "" : value.toString());
            return this;
        }

        public HttpMap clear() {
            map.clear();
            return this;
        }

        @Override
        public String toString() {
            StringBuilder builder = new StringBuilder();
            for (Map.Entry<String, ?> entry : map.entrySet()) {
                try {
                    builder.append(URLEncoder.encode(entry.getKey(), "utf-8"))
                            .append("=")
                            .append(URLEncoder.encode(entry.getValue().toString(), "utf-8"))
                            .append("&");
                } catch (UnsupportedEncodingException e) {
                    e.printStackTrace();
                }
            }
            if (builder.length() > 0) {
                builder.deleteCharAt(builder.length() - 1);
            }
            return builder.toString();
        }
    }
}

File: lib/src/main/java/com/zwb/commonlibs/http/HttpHelper.kt
package com.zwb.commonlibs.http

import com.zwb.commonlibs.utils.LogUtils
import java.io.IOException
import java.io.InputStream
import java.net.HttpURLConnection
import java.net.URL
import java.nio.charset.Charset

@Throws(IOException::class)
fun inputStreamToString(inputStream: InputStream, encoding: String): String {
    return inputStream.bufferedReader(Charset.forName(encoding)).use { it.readText() }
}

fun buildUrl(url: String, map: HttpMap?): String {
    return if (map !== null) url + "?" + map else url
}

/**
 * Created by zhuwenbo on 2017/10/27.
 * 基于URLConnection的Http实现
 */
open class HttpHelper {
    private var encoding: String = "utf-8"

    @Throws(IOException::class)
    fun getGetResult(url: String): String {
        val string = inputStreamToString(getGetInputStream(url), encoding)
        LogUtils.i(this, string)
        return string
    }

    @Throws(IOException::class)
    open fun getGetResult(url: String, map: HttpMap): String {
        return getGetResult(buildUrl(url, map))
    }

    @Throws(IOException::class)
    fun getGetInputStream(url: String): InputStream {
        val urlConnection = URL(url).openConnection() as HttpURLConnection
        setupURLConnection(urlConnection)
        urlConnection.connect()
        return urlConnection.inputStream
    }

    @Throws(IOException::class)
    open fun getPostResult(url: String, data: HttpMap): String {
        val string = inputStreamToString(getPostInputStream(url, data), encoding)
        LogUtils.i(this, string)
        return string
    }

    @Throws(IOException::class)
    fun getPostInputStream(url: String, data: HttpMap?): InputStream {
        val urlConnection = URL(url).openConnection() as HttpURLConnection
        setupURLConnection(urlConnection)
        if (data != null) {
            urlConnection.doOutput = true
            urlConnection.doInput = true
            urlConnection.outputStream.write(data.toString().toByteArray())
            urlConnection.outputStream.flush()
            urlConnection.outputStream.close()
        }
        return urlConnection.inputStream
    }

    open protected fun setupURLConnection(urlConnection: HttpURLConnection) {
        urlConnection.setRequestProperty("content-type", "application/x-www-form-urlencoded")
        urlConnection.connectTimeout = 10000
        urlConnection.readTimeout = 10000
    }
}
